Atomic/backends/_ostree.py

```diff
@@ -165,13 +165,10 @@ def delete_container(self, container, force=False):
 
     def run(self, iobject, **kwargs):
         args = kwargs.get('args')
-        try:
-            name = iobject.name
-        except AttributeError:  # iobject isn't populated
-            raise ValueError('Unable to find an image named {} in {}'.format(
-                args.image, args.storage))
+        name = args.image
```

Comment:
Hmm, I am a bit confused about here. We do have atomic run --name where it requires a existing container. I am slightly concerned that user will do something like `atomic run --storage ostree --name firewalld` where firewalld is the container for example. And no args.image is specified, then will there be a problem?

```
if self.name:
             try:
                 be, con_obj = be_utils.get_backend_and_container_obj(self.name)
                 return be.run(con_obj, atomic=self, args=self.args)
             except ValueError:
pass
```

Reply:
this command was designed for the Docker backend, and containers having labels.  It doesn't fully fit with the way system containers work.  A system container knows the image it was installed from, so we don't need to duplicate this information on the command line.

I think the best way is to raise an error when `--name` is used with the ostree backend.

Reply:
`atomic run --storage ostree --name firewalld` will fail immediately as you don't specify the image.

Reply:
Hmm, sorry, I did not fully understand this implementation. Let me try explaining my POV about this.  I believe once we install the system container, it will be there unless you uninstall it.(and user wants to use it) The usual way we start the system container is via systemctl start container. 

However, some system containers(or in the future) behaves not like that. One example is firewalld, and we can have the following:
1: We install a system container to the system. E.g: firewalld for example
2: We start the service using systemctl start firewalld
3: If we disable the name option, can we still do the following? Like atomic run --name firewalld firewall-cmd xxx? 
(because we want to fully utilize the container once it is installed onto the host right?)

In a sense, those two options (install a container) or run one-shot method are conflicting with each other in this case no? Because, with `atomic run $syscontainer` disabled, then the users who install containers(e.g firewalld) to the host can not access inner commands like firewall-cmd easily.They can only do one-shot option for these system container or run `runc firewalld firewall-cmd` command manually. 

 So my doubt is do we also want to enable the running via "atomic run container" path as well? I am fine with only using one-shot method for this type of system container too, but that would require a broader discussion? (IMO tho). Again, I might misunderstand this part, so feel free to correct me if I was wrong :).

Reply:
yes, this patch doesn't change what we can currently do.  We are still able to do `atomic run --name firewalld firewall-cmd xxx`.

The argument is considered an image when it is either fully specified or contains a `:`, in that case it cannot be a container as it is not a valid container name.

We surely need to figure out a way to make this clear in the `atomic run` documentation and `--help` output, as we already had this problem before, as the command says IMAGE but we were using it as a container name.

Reply:
>atomic run --name firewalld firewall-cmd xxx

But that will default to docker container only right?( part of my long question above was to ask should we have this working as well)
 the following would no longer work: 
`atomic run --name firewalld --storage ostree firewall-cmd xxx` and I tried the patch a little:

```
[root@localhost atomic]# atomic containers list -a | grep firewalld
   firewalld    firewalld:latest     firewalld  /usr/bin/r 2018-02-16 14:49 inactive   ostree     /usr/bin/r
[root@localhost atomic]# atomic run -n firewalld --storage ostree firewall-cmd status
--name is not supported by this backend
```

> The argument is considered an image when it is either fully specified or contains a :, in that case it cannot be a container as it is not a valid container name.

Yea, I understand that part. Before, I was concerned about not being able to run system container via atomic run --storage ostree $container.

> We surely need to figure out a way to make this clear in the atomic run documentation and --help output, as we already had this problem before, as the command says IMAGE but we were using it as a container name.

Yup, documentation definitely helps ;)